domain/src/test/groovy/com/github/maleksandrowicz93/educational/institution/EducationalInstitutionSpec.groovy

```diff
@@ -1,158 +1,676 @@
+    def educationalInstitution = EducationalInstitution.from(snapshot, eventsPublisher)
+    @Shared
+    def facultySetup = facultySetup()
+    @Shared
+    def faculty = educationalInstitution.createFaculty(facultySetup)
+
+    def eventsPublisher = Mock(EventsPublisher)
+
+    def setup() {
+        snapshot = educationalInstitutionSnapshot()
+        educationalInstitution = EducationalInstitution.from(snapshot, eventsPublisher)
+        facultySetup = facultySetup()
+        faculty = educationalInstitution.createFaculty(facultySetup)
+    }
+
     def "faculty may be created"() {
         given: "Educational Institution without any faculty"
-        and: "new faculty to be created"
+        educationalInstitution = EducationalInstitution.from(snapshot, eventsPublisher)
+
         when: "the faculty is created"
+        faculty = educationalInstitution.createFaculty(facultySetup)
+
         then: "should be created successfully"
+        def currentSnapshot = educationalInstitution.createSnapshot()
+        !currentSnapshot.faculties().isEmpty()
+        faculty.id().value() != null
+        faculty.name() == facultySetup.name()
+        validateMainFieldOfStudy(faculty, facultySetup)
+        validateSecondaryFieldsOfStudy(faculty, facultySetup)
+    }
+
+    private void validateMainFieldOfStudy(FacultySnapshot facultySnapshot, FacultySetup facultySetup) {
+        def mainFieldOfStudy = facultySnapshot.mainFieldOfStudy()
+        assert mainFieldOfStudy.id().value() != null
+        assert mainFieldOfStudy.facultyId() == facultySnapshot.id()
+        assert mainFieldOfStudy.name() == facultySetup.mainFieldOfStudyName().value()
+    }
+
+    private void validateSecondaryFieldsOfStudy(FacultySnapshot facultySnapshot, FacultySetup facultySetup) {
+        facultySnapshot.secondaryFieldsOfStudy().forEach {
+            validateSecondaryFieldOfStudy(it, facultySnapshot.id(), facultySetup)
+        }
+    }
+
+    private void validateSecondaryFieldOfStudy(
+            FieldOfStudySnapshot fieldOfStudySnapshot,
+            FacultyId facultyId,
+            FacultySetup facultySetup
+    ) {
+        assert fieldOfStudySnapshot.id().value() != null
+        assert fieldOfStudySnapshot.facultyId() == facultyId
+        assert facultySetup.secondaryFieldsOfStudyNames().stream()
+                .map { it.value() }
+                .collect(toSet())
+                .contains(fieldOfStudySnapshot.name())
     }
 
     def "professor matching all requirements may be employed"() {
         given: "Educational Institution with vacancy at a faculty"
         and: "professor's application matching all requirements"
+        def application = basicProfessorApplication(faculty)
+
         when: "professor applies for hiring"
+        def maybeHired = educationalInstitution.considerHiring(application)
+
         then: "professor should be employed"
+        def professors = getProfessors()
+        !professors.isEmpty()
+        maybeHired.isPresent()
```

Comment:
nie czuję tego API za bardzo - jak jest empty, to dlaczego? jaki jest powód odrzucenia kandydatury?